types/mui-datatables/index.d.ts

```diff
@@ -109,21 +114,27 @@ export interface MUIDataTableFilterOptions {
     logic?: (prop: string, filterValue: any[]) => boolean;
 }
 
+export interface MUIDataTableColumnState extends MUIDataTableColumnOptions {
+    name: string;
+    label?: string;
+}
+
 export interface MUIDataTableColumnOptions {
     customBodyRender?: (value: any, tableMeta: MUIDataTableMeta, updateValue: (s: any, c: any, p: any) => any) => string | React.ReactNode;
     customHeadRender?: (columnMeta: MUIDataTableCustomHeadRenderer, updateDirection: (params: any) => any) => string | React.ReactNode;
     display?: 'true' | 'false' | 'excluded';
     download?: boolean;
     empty?: boolean;
     filter?: boolean;
+    filterType?: FilterType;
     filterList?: string[];
     filterOptions?: MUIDataTableFilterOptions;
     hint?: string;
     print?: boolean;
     searchable?: boolean;
     setCellProps?: (cellValue: string, rowIndex: number, columnIndex: number) => object;
     sort?: boolean;
-    sortDirection?: 'asc' | 'desc';
+    sortDirection?: 'asc' | 'desc' | 'none'; // TODO why 'none' might be on this field in practice? it should be just undefined. and it should reuse SortDirection type
```

Comment:
```suggestion
    sortDirection?: 'asc' | 'desc' | 'none';
```

Because the library has defined `none` we type it as such. If you want to get `undefined` in here create an issue on the library's repo. Library authors have no obligations to check this repo or update their code to make it better for TS user's wishes (as nice as it would be...)

reference: https://github.com/gregnb/mui-datatables/#column-options

Reply:
Done!